### ISSUE TITLE
精度达不到论文里面的数据

### ISSUE THREAD

**Issue by zqx951102** [WangTaoAs/PFD_Net#3]
作者您好，我在1501上测试了一下 就改了
/home/zqx_3090/PersonReID/PersonReID2/PFD_Net-master/configs/Market1501/skeleton_pfd.yml
这个文件，里面的参数并没有改动  改了权重的路径，和文件夹的路径 其他都没变，如何训练300轮次后 我选择最高300轮的
/home/zqx_3090/PersonReID/PersonReID2/PFD_Net-master/logs/Market/pfd_net/skeleton_transformer_300.pth
去测试  结果是 ：
2021-12-28 18:23:39,417 PFDreid.test INFO: Validation Results 
2021-12-28 18:23:39,417 PFDreid.test INFO: mAP: 88.2%
2021-12-28 18:23:39,418 PFDreid.test INFO: CMC curve, Rank-1  :94.8%
2021-12-28 18:23:39,418 PFDreid.test INFO: CMC curve, Rank-5  :98.3%
2021-12-28 18:23:39,418 PFDreid.test INFO: CMC curve, Rank-10 :99.0%
达不到论文的95.5  甚至不如TransReID的精度 ？？？ 您能看看是为什么嘛？

MODEL:
  PRETRAIN_CHOICE: 'imagenet'
  PRETRAIN_PATH: '/home/zqx_3090/PersonReID/PersonReID2/PFD_Net-master/weights/jx_vit_base_p16_224-80ecf9dd.pth'
  METRIC_LOSS_TYPE: 'triplet'
  IF_LABELSMOOTH: 'on'
  IF_WITH_CENTER: 'no'
  NAME: 'skeleton_transformer'
  NO_MARGIN: True
  DEVICE_ID: ('2')
  TRANSFORMER_TYPE: 'vit_base_patch16_224_TransReID'
  STRIDE_SIZE: [16, 16]

  SIE_CAMERA: True
  SIE_COE: 3.0
  JPM: True
  RE_ARRANGE: True
  NUM_HEAD: 8
  DECODER_DROP_RATE: 0.1
  DROP_FIRST: False
  NUM_DECODER_LAYER: 6
  QUERY_NUM: 17
  POSE_WEIGHT: '/home/zqx_3090/PersonReID/PersonReID2/PFD_Net-master/weights/pose_hrnet_w48_384x288.pth'
  SKT_THRES: 0.2
  
INPUT:
  SIZE_TRAIN: [256, 128]
  SIZE_TEST: [256, 128]
  PROB: 0.5 # random horizontal flip
  RE_PROB: 0.5 # random erasing
  PADDING: 10
  PIXEL_MEAN: [0.5, 0.5, 0.5]
  PIXEL_STD: [0.5, 0.5, 0.5]

DATASETS:
  NAMES: ('market1501')
  ROOT_DIR: ('/home/zqx_3090/PersonReID/PersonReID2/PFD_Net-master/data/')

DATALOADER:
  SAMPLER: 'softmax_triplet'
  NUM_INSTANCE: 4
  NUM_WORKERS: 8

SOLVER:
  OPTIMIZER_NAME: 'SGD'
  MAX_EPOCHS: 300
  BASE_LR: 0.008
  IMS_PER_BATCH: 64
  WARMUP_METHOD: 'linear'
  LARGE_FC_LR: False
  CHECKPOINT_PERIOD: 60
  LOG_PERIOD: 50
  EVAL_PERIOD: 30
  WEIGHT_DECAY:  1e-4
  WEIGHT_DECAY_BIAS: 1e-4
  BIAS_LR_FACTOR: 2

TEST:
  EVAL: True
  IMS_PER_BATCH: 256
  RE_RANKING: False
  WEIGHT: "/home/zqx_3090/PersonReID/PersonReID2/PFD_Net-master/logs/Market/pfd_net/skeleton_transformer_300.pth" #put your own pth
  NECK_FEAT: 'before'
  FEAT_NORM: 'yes'

OUTPUT_DIR: 'logs/Market/pfd_net'

**Comment by zqx951102** [comment]
而且 我发现 训练精度和测试精度的 输出一模一样  这又是为什么呢？

**Comment by WangTaoAs** [comment]
> 作者您好，我在1501上测试了一下 就改了 /home/zqx_3090/PersonReID/PersonReID2/PFD_Net-master/configs/Market1501/skeleton_pfd.yml 这个文件，里面的参数并没有改动 改了权重的路径，和文件夹的路径 其他都没变，如何训练300轮次后 我选择最高300轮的 /home/zqx_3090/PersonReID/PersonReID2/PFD_Net-master/logs/Market/pfd_net/skeleton_transformer_300.pth 去测试 结果是 ： 2021-12-28 18:23:39,417 PFDreid.test INFO: Validation Results 2021-12-28 18:23:39,417 PFDreid.test INFO: mAP: 88.2% 2021-12-28 18:23:39,418 PFDreid.test INFO: CMC curve, Rank-1 :94.8% 2021-12-28 18:23:39,418 PFDreid.test INFO: CMC curve, Rank-5 :98.3% 2021-12-28 18:23:39,418 PFDreid.test INFO: CMC curve, Rank-10 :99.0% 达不到论文的95.5 甚至不如TransReID的精度 ？？？ 您能看看是为什么嘛？
> 
> MODEL: PRETRAIN_CHOICE: 'imagenet' PRETRAIN_PATH: '/home/zqx_3090/PersonReID/PersonReID2/PFD_Net-master/weights/jx_vit_base_p16_224-80ecf9dd.pth' METRIC_LOSS_TYPE: 'triplet' IF_LABELSMOOTH: 'on' IF_WITH_CENTER: 'no' NAME: 'skeleton_transformer' NO_MARGIN: True DEVICE_ID: ('2') TRANSFORMER_TYPE: 'vit_base_patch16_224_TransReID' STRIDE_SIZE: [16, 16]
> 
> SIE_CAMERA: True SIE_COE: 3.0 JPM: True RE_ARRANGE: True NUM_HEAD: 8 DECODER_DROP_RATE: 0.1 DROP_FIRST: False NUM_DECODER_LAYER: 6 QUERY_NUM: 17 POSE_WEIGHT: '/home/zqx_3090/PersonReID/PersonReID2/PFD_Net-master/weights/pose_hrnet_w48_384x288.pth' SKT_THRES: 0.2
> 
> INPUT: SIZE_TRAIN: [256, 128] SIZE_TEST: [256, 128] PROB: 0.5 # random horizontal flip RE_PROB: 0.5 # random erasing PADDING: 10 PIXEL_MEAN: [0.5, 0.5, 0.5] PIXEL_STD: [0.5, 0.5, 0.5]
> 
> DATASETS: NAMES: ('market1501') ROOT_DIR: ('/home/zqx_3090/PersonReID/PersonReID2/PFD_Net-master/data/')
> 
> DATALOADER: SAMPLER: 'softmax_triplet' NUM_INSTANCE: 4 NUM_WORKERS: 8
> 
> SOLVER: OPTIMIZER_NAME: 'SGD' MAX_EPOCHS: 300 BASE_LR: 0.008 IMS_PER_BATCH: 64 WARMUP_METHOD: 'linear' LARGE_FC_LR: False CHECKPOINT_PERIOD: 60 LOG_PERIOD: 50 EVAL_PERIOD: 30 WEIGHT_DECAY: 1e-4 WEIGHT_DECAY_BIAS: 1e-4 BIAS_LR_FACTOR: 2
> 
> TEST: EVAL: True IMS_PER_BATCH: 256 RE_RANKING: False WEIGHT: "/home/zqx_3090/PersonReID/PersonReID2/PFD_Net-master/logs/Market/pfd_net/skeleton_transformer_300.pth" #put your own pth NECK_FEAT: 'before' FEAT_NORM: 'yes'
> 
> OUTPUT_DIR: 'logs/Market/pfd_net'

----------------------------------------


你好，针对你提出的问题，我在论文中有的实现细节中有讲到，对于occluded-duke这类小的数据集我们采用了2个decoder，对于大型的数据集market之类的，我们采用了6个decoder。
![image](https://user-images.githubusercontent.com/52031839/147561383-60fc7370-bdb6-4fe5-9f48-2915ff5e37b6.png)
其次，根据你的log显示是在3090上进行的训练，我们之前也在3090上训练过，但确实没有达到我们1080ti上的效果（目前还不知道原因，可能是cuda11的问题？）
你可以试着调整参数～

**Comment by zqx951102** [comment]
好的 十分感谢作者您的回复。我再试着调一下参数. 关于3090这个问题 确实有可能是cuda11 但是按照咱们常识理解。这个gpu 应该不会影响精度 只会影响训练速度啊 等我后续换别的gpu试一下 再告诉您tesla上的效果 不过实验室服务器好像都是cuda11版本的 &nbsp;最后十分感谢您的回信





------------------ 原始邮件 ------------------
发件人: Tao Wang ***@***.***&gt;
发送时间: 2021年12月28日 19:25
收件人: WangTaoAs/PFD_Net ***@***.***&gt;
抄送: zqx951102 ***@***.***&gt;, Author ***@***.***&gt;
主题: Re: [WangTaoAs/PFD_Net] 精度达不到论文里面的数据 (Issue #3)





  
作者您好，我在1501上测试了一下 就改了 /home/zqx_3090/PersonReID/PersonReID2/PFD_Net-master/configs/Market1501/skeleton_pfd.yml 这个文件，里面的参数并没有改动 改了权重的路径，和文件夹的路径 其他都没变，如何训练300轮次后 我选择最高300轮的 /home/zqx_3090/PersonReID/PersonReID2/PFD_Net-master/logs/Market/pfd_net/skeleton_transformer_300.pth 去测试 结果是 ： 2021-12-28 18:23:39,417 PFDreid.test INFO: Validation Results 2021-12-28 18:23:39,417 PFDreid.test INFO: mAP: 88.2% 2021-12-28 18:23:39,418 PFDreid.test INFO: CMC curve, Rank-1 :94.8% 2021-12-28 18:23:39,418 PFDreid.test INFO: CMC curve, Rank-5 :98.3% 2021-12-28 18:23:39,418 PFDreid.test INFO: CMC curve, Rank-10 :99.0% 达不到论文的95.5 甚至不如TransReID的精度 ？？？ 您能看看是为什么嘛？
 
MODEL: PRETRAIN_CHOICE: 'imagenet' PRETRAIN_PATH: '/home/zqx_3090/PersonReID/PersonReID2/PFD_Net-master/weights/jx_vit_base_p16_224-80ecf9dd.pth' METRIC_LOSS_TYPE: 'triplet' IF_LABELSMOOTH: 'on' IF_WITH_CENTER: 'no' NAME: 'skeleton_transformer' NO_MARGIN: True DEVICE_ID: ('2') TRANSFORMER_TYPE: 'vit_base_patch16_224_TransReID' STRIDE_SIZE: [16, 16]
 
SIE_CAMERA: True SIE_COE: 3.0 JPM: True RE_ARRANGE: True NUM_HEAD: 8 DECODER_DROP_RATE: 0.1 DROP_FIRST: False NUM_DECODER_LAYER: 6 QUERY_NUM: 17 POSE_WEIGHT: '/home/zqx_3090/PersonReID/PersonReID2/PFD_Net-master/weights/pose_hrnet_w48_384x288.pth' SKT_THRES: 0.2
 
INPUT: SIZE_TRAIN: [256, 128] SIZE_TEST: [256, 128] PROB: 0.5 # random horizontal flip RE_PROB: 0.5 # random erasing PADDING: 10 PIXEL_MEAN: [0.5, 0.5, 0.5] PIXEL_STD: [0.5, 0.5, 0.5]
 
DATASETS: NAMES: ('market1501') ROOT_DIR: ('/home/zqx_3090/PersonReID/PersonReID2/PFD_Net-master/data/')
 
DATALOADER: SAMPLER: 'softmax_triplet' NUM_INSTANCE: 4 NUM_WORKERS: 8
 
SOLVER: OPTIMIZER_NAME: 'SGD' MAX_EPOCHS: 300 BASE_LR: 0.008 IMS_PER_BATCH: 64 WARMUP_METHOD: 'linear' LARGE_FC_LR: False CHECKPOINT_PERIOD: 60 LOG_PERIOD: 50 EVAL_PERIOD: 30 WEIGHT_DECAY: 1e-4 WEIGHT_DECAY_BIAS: 1e-4 BIAS_LR_FACTOR: 2
 
TEST: EVAL: True IMS_PER_BATCH: 256 RE_RANKING: False WEIGHT: "/home/zqx_3090/PersonReID/PersonReID2/PFD_Net-master/logs/Market/pfd_net/skeleton_transformer_300.pth" #put your own pth NECK_FEAT: 'before' FEAT_NORM: 'yes'
 
OUTPUT_DIR: 'logs/Market/pfd_net'
  
 
你好，针对你提出的问题，我在论文中有的实现细节中有讲到，对于occluded-duke这类小的数据集我们采用了2个decoder，对于大型的数据集market之类的，我们采用了6个decoder。
 
 其次，根据你的log显示是在3090上进行的训练，我们之前也在3090上训练过，但确实没有达到我们1080ti上的效果（目前还不知道原因，可能是cuda11的问题？）
 还有就是这套参数是针对occluded数据集的，你可以试着调整参数～
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
